js/components/graph/Graph.js

```diff
@@ -338,6 +341,77 @@ export default class Graph extends React.Component {
     }
   };
 
+  /**
+   * Returns the position of the mouse adjusted to the viewbox's new origin. This is so that the change
+   * in viewboxX and viewboxY are relative to the shifted viewbox origin after panning.
+   * @param {double} clientX
+   * @param {double} clientY
+   * @return {Object}
+   */
+  adjustCoordsToViewbox = (clientX, clientY) => {
+    return {
+      adjustedX: clientX + this.state.viewBoxPos.x,
+      adjustedY: clientY + this.state.viewBoxPos.y
+    };
+  }
+
+  /**
+   * Initializes the panning process by recording the position of the mouse pointer and adding event listeners.
+   * @param {*} event
+   */
+  onMouseDownInGraph = event => {
+    document.body.style.cursor = "grab";
+    const {adjustedX, adjustedY} = this.adjustCoordsToViewbox(event.clientX, event.clientY);
+
+    document.getElementById("react-graph").addEventListener("mouseup", this.onMouseUpInGraph);
+    document.getElementById("react-graph").addEventListener("mousemove", this.onMouseMoveInGraph);
+
+    this.setState({
+      mouseDown: true,
+      panning: true,
+      panStartX: adjustedX,
+      panStartY: adjustedY
+    });
+  }
+
+  /**
+   * Pans the graph by moving it in the direction that the mouse moved.
+   * @param {*} event
+   */
+  onMouseMoveInGraph = event => {
```

Comment:
It is better to use method names that reflect the logical behaviour of the method, not the context you expect it to be called from. So for example, something like `panGraph` is better since it describes the action being taken.